tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -44,6 +45,11 @@ setup() {
         "io.containerd.cri.runtime-handler" \
         "kata-${KATA_HYPERVISOR}"
 
+    [[ " ${SUPPORTED_NON_TEE_HYPERVISORS} " =~ " ${KATA_HYPERVISOR} " ]] && \
+        set_metadata_annotation "$kata_pod_with_nydus_config" \
+            "io.katacontainers.config.hypervisor.image" \
+            "/opt/kata/share/kata-containers/kata-containers-confidential.img"
```

Comment:
Should we move this block into `confidential_setup` replacing the `info "Need to apply image annotations"` line?

Reply:
I dislike that boilerplate all over the tests, but it cannot be moved to `confidential_setup`. I explain: at the time `confidential_setup()` is called (in `setup()`) the `"$kata_pod_with_nydus_config"` file wasn't generated yet.  The config file is created with a call to `new_pod_config()` from the test method.

We should revisit and refactor the tests though.

@fidencio as part of his efforts to disable `shared_fs`, hacked the [setup.sh](https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/setup.sh) to set the `cri-handler` on all [runtimeclass_workloads](https://github.com/kata-containers/kata-containers/tree/main/tests/integration/kubernetes/runtimeclass_workloads). I think this is the way to go for setting the image annotation when running on non-TEE. However, k8s-guest-pull-image.bats and k8s-measured-rootfs.bats use a different approach from the rest of the bats, which is to create a simple configuration file out of a template by calling `new_pod_config()`. If we ensure that template is proper instrumented by `setup.sh` then I think we are fine. Anyway, waiting @fidencio to publish his code to then think more deeply how to conciliate approaches.

Reply:
Fair enough, I wonder if we could move the config creation as part of the setup, and remove this issue, but we can worry about that once it's more settled.

Thanks for the explaination.

Reply:
Maybe a nontee runtime class would avoid having to make these kind of switches.

Reply:
yes, a nontee runtime class solve that. But probably introduce other problems. It's a trade-off that we should be discussion at some point